docs/tutorial/01_Rasterizer.md

```diff
@@ -52,18 +52,18 @@ public class TutorialWorldRasterizer implements WorldRasterizer {
     }
 
     @Override
-    public void generateChunk(CoreChunk chunk, Region chunkRegion) {
-        for (Vector3iC position : chunkRegion.getRegion()) {
+    public void generateChunk(Chunk chunk, Region chunkRegion) {
+        // reusing one mutable vector is more efficient than creating a new one for each toRelative()
+        Vector3i tmp = new Vector3i();
+        for (Vector3ic position : chunkRegion.getRegion()) {
             if (position.y() < 0) {
-                chunk.setBlock(position, dirt);
+                chunk.setBlock(Chunks.toRelative(position, tmp), dirt);
```

Comment:
I understand why we have the tmp vector here, but I'm finding this is a case where I dislike the readability of the result.

Having `Chunks.toRelative(position, tmp)` as a method that modifies `tmp` is fine.  
And I _know_ the reason it returns itself as the return value is so you can chain other vector method calls on the end of it.

But gosh, to the casual reader it looks for all the world like we're just passing in some uninitialized variable there, or writing something to that object but then never _using_ that result.

Reply:
I guess one alternative would be
```java
Chunks.toRelative(position, relativePosition)
chunk.setBlock(relativePosition, dirt)
```

We could also take advantage of the fact that this rasterizer is so simple to come up with other alternatives, but I guess since this example is a starting place to build other things on, that's a reason for it to keep using this iterate-over-every-position technique.

Reply:
That's valid feedback here :+1: I'm not sure about the best way to write this kind of code, so having somebody read and understand it who is not familiar with it helps a lot.

Another alternative to what you proposed is to store the current position in `tmp` more explicitly, and then to the computation of `toRelative` only on a single argument:

```java
tmp.set(position);
chunk.setBlock(Chunks.toRelative(tmp), dirt);
```

Your suggestion could be made even more verbose here by adding in another assignment (where I hope it gets optimized away by the JIT compiler):

```java
relativePosition = Chunks.toRelative(position, relativePosition);
chunk.setBlock(relativePosition, dirt);
```